### PR TITLE
Maintenance updates

### LINKED ISSUE #105: CI: prefer ubuntu-22.04 to ubuntu-latest
Or otherwise start CI'ing only using Ubuntu versions [mentioned here](https://repo.hex.pm/builds/otp/ubuntu-22.04/builds.txt) (valid until Ubuntu latest becomes something else again).

### LINKED ISSUE #106: softlink in _checkouts folder?
Hi,

As I see the examples folder contains a soft link to ../../. : https://github.com/inaka/cowboy-trails/blob/master/example/_checkouts/trails

Using a quite old version of rebar3 (3.13.3) sometimes `rebar3 as prod tar` works, sometimes don't (I don't know what causes the problem, on some machines it works, but on some other Jenkins agents we got this error):

```
 ===> error (eloop) accessing file /var/Jenkins/workspace/Upgrade_Release/our_app/_build/prod/lib/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/example/_checkouts/trails/.github
```
Looks like the problem is also related not only to rebar3 but to the OS / filesystem under the engine.
So basically the softlink pointing back to the root directory causes an infinite loop under some conditions.
Would it be possible to delete this softlink, or was it intentionally committed in : https://github.com/inaka/cowboy-trails/commit/ec2fe6ffc699a99bf2ae8cab8fb2ee9763cf7aad ?

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
This isn't as big as it looks, though there's possibly a lotta changes.

I'll self-review and try to guide you through it to ease the task.

Whenever a change is not super-explicit or supported by the commit message I'll add a comment.

Also closes #105.
Also closes #106.

**Note**: the first commit is a big format change, which explains why the `.erl` (and friends files) changed so much.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,42 +1,36 @@
 ---
 name: build
-on:
-  push:
-    branches:
-      - master
-  pull_request:
-    branches:
-      - master
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:9]: To use pre-pull request CI on forks.

File: example/rebar.config
Comment (paulo-ferraz-oliveira) [inline example/rebar.config:1]: Changes better explained in 4acf05e.

File: example/rel/vm.args
Comment (paulo-ferraz-oliveira) [inline example/rel/vm.args:1]: Without this I was getting warnings and exit codes different from 0 is GitHub CI.

File: rebar.config
Comment (paulo-ferraz-oliveira) [inline rebar.config:1]: Better explain in 4acf05e's message.

File: src/trails.app.src [hunk 2]
```diff
@@ -1,23 +1,15 @@
...
-           {"Example", "https://github.com/inaka/cowboy-trails/tree/master/example"}
-         ]},
-  {build_tools,["rebar3"]}
- ]
-}.
+  {maintainers, ["Inaka"]},
+  {licenses, ["Apache 2.0"]},
+  {links,
+   [{"GitHub", "https://github.com/inaka/cowboy-trails"},
+    {"Blog", "https://web.archive.org/web/20160422045517/http://inaka.net/blog/2015/07/20/cowboy-trails/"},
```
Comment (paulo-ferraz-oliveira) [inline src/trails.app.src:13]: Recovered it from the archive 😄

File: test/trails_SUITE.erl [hunk 3]
```diff
@@ -1,7 +1,5 @@
 -module(trails_SUITE).
 
--author('elbrujohalcon@inaka.net').
-
```
Comment (paulo-ferraz-oliveira) [inline test/trails_SUITE.erl]: Elvis, right? Lemme know if you prefer this to be present elsewhere, e.g. in a comment.

File: src/trails.erl [hunk 4]
```diff
@@ -147,7 +147,7 @@ constraints(Trail) ->
 %%      with `cowboy'). Your handler must implement the callback `c:trails_handler:trails/0'
 %%      and return the specific routes for that handler. That callback is
 %%      invoked for each given module and then the results are concatenated.
--spec trails(module() | [module()]) -> trails().
+-spec trails(module() | [module()] | {module(), map()} | [{module(), map()}]) -> trails().
```
Comment (paulo-ferraz-oliveira) [inline src/trails.erl:150]: Maybe these can be named and written with composed types (?)

**Comment by paulo-ferraz-oliveira** [comment]
There might be some unexpected branch protection rules for this
image

File: src/trails.erl (hunk 4 above)
Comment (elbrujohalcon) [reply src/trails.erl:150]: Yeah, maybe in another PR.

**Review by elbrujohalcon — APPROVED** [review]

File: test/trails_SUITE.erl (hunk 3 above)
Comment (elbrujohalcon) [reply test/trails_SUITE.erl]: Kill it with fire!

**Comment by elbrujohalcon** [comment]
https://hex.pm/packages/trails/2.3.1

**Pull request merged**